### PR TITLE
Journalist can login to create article

### PR DESCRIPTION
**Pull request by ViktorHek**
[Journalist has to be logged in to create an article ADMIN](https://www.pivotaltracker.com/story/show/176339855)
## User Story
```
As a journalist,
In order to restrict the access of creating articles,
I would like to be able to have a login functionality.
```
## Changes proposed in this pull request:
* Creating login test and creating two fixture file with journalist and registered user
* Created a login form
* Authentication action I rootReducer
* Creating authentication model
* Adding ternary operator in app.jsx so that article form only show if user is logged in.
## What I have learned working on this feature:
* j-tock auth
* importance of checking initializers in API
## Packages/Gems added
* j-tock auth
## Screenshots (Client)
Skärmavbild 2021-01-10 kl  21 09 01

### CODE REVIEW

File: cypress/integration/journalistCanLoginAndSeeArticleForm.feature.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    cy.get("[data-cy='login-form']").within(() => {
+      cy.get("[data-cy='email']").type("journalist@mail.com");
+      cy.get("[data-cy='password']").type("password");
+      cy.get("[data-cy='submit-btn']").contains("Submit").click();
+    });
+    cy.get("[data-cy='header-user-email']").should(
+      "contain",
+      "Meow! Welcome back journalist@mail.com"
+    );
+
```
Comment (Ehugo2000) [inline cypress/integration/journalistCanLoginAndSeeArticleForm.feature.js]: Format the code here.

File: cypress/integration/journalistCanLoginAndSeeArticleForm.feature.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      url: "http://localhost:3000/api/auth/sign_in",
+      response: "fixture:registered_user_can_not_log_in.json",
+    });
+    cy.route({
+      method: "GET",
+      url: "http://localhost:3000/api/auth/validate_token**",
+      response: "fixture:registered_user_can_not_log_in.json",
+    });
+    cy.visit("/");
+
```
Comment (Ehugo2000) [inline cypress/integration/journalistCanLoginAndSeeArticleForm.feature.js:57]: Format the testfile, especially for the white spaces, not consistent

File: src/App.jsx [hunk 3]
```diff
@@ -1,12 +1,17 @@
-import React from 'react'
-import CreateArticleForm from './components/CreateArticleForm'
+import React from "react";
+import { useSelector } from "react-redux";
+import CreateArticleForm from "./components/CreateArticleForm";
+import LoginForm from "./components/LoginForm";
+import Header from "./components/Header";
 
 const App = () => {
+  const currentUser = useSelector((state) => state.currentUser);
```
Comment (Ehugo2000) [inline src/App.jsx]: Remove () from `(state)`

File: cypress/integration/journalistCanLoginAndSeeArticleForm.feature.js (hunk 1 above)
Comment (ViktorHek) [reply cypress/integration/journalistCanLoginAndSeeArticleForm.feature.js]: done 👍

File: cypress/integration/journalistCanLoginAndSeeArticleForm.feature.js (hunk 2 above)
Comment (ViktorHek) [reply cypress/integration/journalistCanLoginAndSeeArticleForm.feature.js:57]: fixed

File: src/App.jsx (hunk 3 above)
Comment (ViktorHek) [reply src/App.jsx]: jupp jupp. it's done :)

File: cypress/integration/journalistCanCreateAnArticle.feature.js [hunk 4]
```diff
@@ -1,15 +1,33 @@
-/* eslint-disable no-undef */
 describe("Journalist can create an article", () => {
```
Comment (emtalen) [inline cypress/integration/journalistCanCreateAnArticle.feature.js:1]: please go over how you structure this file, have one describe block wrapping everything and then divide it into two other describe blocks inside where you have happy and sad path. You can still have a before each with logging in the journalist

**Review by emtalen — CHANGES_REQUESTED** [review]

File: cypress/integration/journalistCanCreateAnArticle.feature.js (hunk 4 above)
Comment (ViktorHek) [reply cypress/integration/journalistCanCreateAnArticle.feature.js:1]: we have fixed this now. hope you like in :)

**Review by emtalen — APPROVED** [review]

**Pull request merged**